### ISSUE TITLE
Introduce DDD concept of value objects

### ISSUE THREAD

**Issue by Abdull** [jhipster/jhipster-core#262]
##### **Overview of the feature request**
Domain-driven design (DDD) has the concept of _value objects_ (see https://en.wikipedia.org/wiki/Value_object , https://martinfowler.com/bliki/ValueObject.html ).

Currently, JHipster JDL cannot model value objects.

##### **Motivation for or Use Case**
Value objects are a significant concept when modeling a sound domain following DDD. Other JHipster users have expressed interest in it ( https://stackoverflow.com/q/41082676/923560 , https://stackoverflow.com/q/44657325/923560 ).
Currently, implementing value objects within a JHipster application is a manual task and requires manual modification of generated code, even though IMHO it could be easily automated.

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by murdos** [comment]
+1
Just for reference, #128 was a similar request.

**Comment by cbornet** [comment]
Can we really do something smart here ? IMO, value objects shouldn't be anemic, they should embed the domain business logic. So I'm not sure it is compatible with code generation...

**Comment by MathieuAA** [comment]
Yes Christophe, you're right. However, I can see this happening if someone makes a blueprint first

**Comment by jdubois** [comment]
Totally agree with @cbornet on this - for me, if you want this design (which has some big drawbacks but that's not the point), you just use your IDE to refactor the generated code.
